### PR TITLE
Websocket support for teleportation

### LINKED ISSUE #993: Consider adding WS instead of Polling events (Teleporter)
## Issue Type

```
[ ] bug report
[ X] feature request
```

## Current Behavior
eth_subscribe for ws has been rolled out

TBD

### PR DESCRIPTION
**Pull request by wsdt**
:clipboard: Resolves #993

## Overview

Added Websocket support Teleportation.

## Changes

- Added WS specific integration test
- Teleportation service uses by default Websocket if provided, if not it falls back to previous http polling strategy
- Websocket and http providers can be mixed, as there might not be a websocket for every supported network. 
- Websocket checks via http first for previous events based on saved lastBlock.

## Testing

- Websocket tests run through
- Previous tests work too (for both httpProvider and wsProvider)

### CODE REVIEW

File: packages/boba/teleportation/src/service.ts [hunk 1]
```diff
@@ -143,8 +154,31 @@ export class TeleportationService extends BaseService<TeleportationOptions> {
...
+      const events: AssetReceivedEvent[] = await this._watchTeleportation(
+        depositTeleportation,
+        latestBlock
+      )
+      await this._disburseTeleportation(
+        depositTeleportation,
+        events,
+        latestBlock
+      )
+
```
Comment (souradeep-das) [inline packages/boba/teleportation/src/service.ts]: is a try/catch required here?

**Review by souradeep-das — APPROVED** [review]
lgtm!

File: packages/boba/teleportation/src/service.ts (hunk 1 above)
Comment (wsdt) [reply packages/boba/teleportation/src/service.ts]: good one, doesn't harm

File: packages/boba/teleportation/src/utils/chains.ts [hunk 2]
```diff
@@ -21,6 +25,7 @@ export const BobaChains: IBobaChains = {
   //#region boba_networks
   288: {
     url: 'https://replica.boba.network',
+    // wsUrl: 'wss://boba-ethereum.gateway.tenderly.co',
```
Comment (wsdt) [inline packages/boba/teleportation/src/utils/chains.ts:28]: not using ws for now

**Comment by InoMurko** [comment]
Closed for now as it's not needed, we can bring it back if it becomes a neccessety

**Pull request closed without merging**